app/controllers/spree/admin/variants_controller.rb

```diff
@@ -4,14 +4,51 @@ module Spree
   module Admin
     class VariantsController < ResourceController
       helper 'spree/products'
+      include ProductFilterHelper
+
       belongs_to 'spree/product', find_by: :permalink
       new_action.before :new_before
 
+      def index
+        @url_filters = product_filters(request.query_parameters)
+      end
+
+      def edit
+        @url_filters = product_filters(request.query_parameters)
+
+        super
+      end
+
+      def update
+        @url_filter = product_filters(request.query_parameters)
+
+        if @object.update(permitted_resource_params)
+          flash[:success] = flash_message_for(@object, :successfully_updated)
+          redirect_to admin_product_variants_url(params[:product_id], @url_filter)
+        else
+          redirect_to edit_admin_product_variant_url(params[:product_id], @object, @url_filter)
+        end
+      end
+
+      def new
+        @url_filters = product_filters(request.query_parameters)
+
+        super
```

Comment:
same here, instead of super, probably a one liner using render

Reply:
fixed